openshift-ci/README.md

```diff
@@ -0,0 +1,193 @@
+# DevConsole Operator Continuous Integration
+
+DevConsole operator uses [OpenShift CI][openshift-ci] for continuous
+integration.  The Openshift CI is built using
+[CI-Operator][ci-operator].  The configuration is located here:
+http://bit.ly/304kpIo
+
+As part of the continuous integration, there are several
+jobs configured to run against pull requests in GitHub.  The CI jobs
+are triggered whenever there is a new pull request from the team
+members.  It is also triggered when there is a new commit in the
+current pull request.  If a non-member creates the pull request, there
+should be a comment with text as `/ok-to-test` from any member to run
+the CI job.
+
+Here is a high-level schematic diagram of the CI pipeline for an
+end-to-end test:
+
+```
++--------+     +--------+     +--------+
+|        |     |        |     |        |
+|  root  +---->+  src   +---->+  bin   |
+|        |     |        |     |        |
++--------+     +--------+     +---+----+
+                                   |
+    ,------------------------------'
+    |
+    v
++---+----+     +--------+
+|        |     |        |
+| images +---->+ tests  |
+|        |     |        |
++--------+     +--------+
+```
+
+For lint and unit test, the schematic diagram is as follows:
+
+```
++--------+     +--------+     +----------------+
+|        |     |        |     |                |
+|  root  +---->+  src   +---->+ lint/unit test |
+|        |     |        |     |                |
++--------+     +--------+     +----------------+
+```
+
+
+All the steps mentioned here is taking place inside a work namespace.
+When you click on the job details from your pull request, you can see
+the name of the work namespace at top.  The name is going start with
+`ci-op-`.  The images created here will be available under this
+namespace.  The namespace can be accessed through
+`OPENSHIFT_BUILD_NAMESPACE` environment variable.
+
+## root
+
+As part of the CI pipeline, the first step is to create `root` image.
+In fact, `root` is a tag created for the pipeline image.  This image
+contains all the tools including but not limited to Go compiler, git,
+kubectl, oc, and Operator SDK.
+
+The `root` image tag is created using this Dockerfile:
+`openshift-ci/Dockerfile.tools`
+
+## src
+
+This step clone the pull request branch with latest changes.  The
+cloning is taking place inside a container created from the `root`
+image.  As a result of this step an image named `src` is going to be
+created.
+
+In the CI configuration, there is a declaration like this:
+
+```
+canonical_go_repository: github.com/redhat-developer/devconsole-operator
+```
+
+The above line ensure the cloned source code goes into the specified
```

Comment:
Small nit - should be: "The above line ensures"

Reply:
Done. ceba1627b1c5d4a8ced946a3b352b1416f4e6c3a